src/tools/clippy/clippy_lints/src/ranges.rs

```diff
@@ -322,18 +322,18 @@ fn check_range_zip_with_len(cx: &LateContext<'_>, path: &PathSegment<'_>, args:
         if path.ident.as_str() == "zip";
         if let [iter, zip_arg] = args;
         // `.iter()` call
-        if let ExprKind::MethodCall(iter_path, _, iter_args, _) = iter.kind;
+        if let ExprKind::MethodCall(ref iter_path, _, ref iter_args, _) = iter.kind;
```

Comment:
Usually `ref` or `&` is unnecessary with HIR since everything is already a reference. This creates double references.